protocol/storage/miner.go

```diff
@@ -619,31 +621,78 @@ func (sm *Miner) currentProvingPeriodPoStChallengeSeed(ctx context.Context) (pro
 	return seed, nil
 }
 
-// OnNewHeaviestTipSet is a callback called by node, everytime the the latest head is updated.
-// It is used to check if we are in a new proving period and need to trigger PoSt submission.
-func (sm *Miner) OnNewHeaviestTipSet(ts types.TipSet) {
-	ctx := context.Background()
+// isBootstrapMinerActor is a convenience method used to determine if the miner
+// actor was created when bootstrapping the network. It it was,
+func (sm *Miner) isBootstrapMinerActor(ctx context.Context) (bool, error) {
+	returnValues, sig, err := sm.porcelainAPI.MessageQuery(
+		ctx,
+		address.Address{},
+		sm.minerAddr,
+		"isBootstrapMiner",
+	)
```

Comment:
Does `gengen` already add this message? I'm not too familiar with the messaging code, but I'd assume that this message needs to exist in the genesis block.

Reply:
Oh wait, I think I just understood what @phritz was saying a while back about MessageQuery and MessageSend being very similar. My guess is that this is just going to call that new method you added.

Reply:
> My guess is that this is just going to call that new method you added.

Yes indeedly